x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/ReplaceSourceAttributes.java

```diff
@@ -44,7 +44,17 @@ protected PhysicalPlan rule(EsSourceExec plan) {
             }
             return new EsQueryExec(plan.source(), plan.index(), plan.indexMode(), List.of(docId, tsid, timestamp), plan.query());
         } else {
-            return new EsQueryExec(plan.source(), plan.index(), plan.indexMode(), List.of(docId), plan.query());
+            // extract and pass _score metadata attribute, if present
+            Attribute scoring = null;
+            for (Attribute attr : plan.output()) {
+                String name = attr.name();
+                if (name.equals(MetadataAttribute.SCORE)) {
```

Comment:
Is this sufficient to detect whether `_score` is set from `metadata`? Some processing commands can create new `Attribute`, like `eval`, `rename` etc.

Reply:
I was not aware, I think you're right, I'll make sure that's not just an attribute name check.